src/OrbitQt/orbitmainwindow.cpp

```diff
@@ -1130,25 +1124,32 @@ bool OrbitMainWindow::eventFilter(QObject* watched, QEvent* event) {
   return QObject::eventFilter(watched, event);
 }
 
-bool OrbitMainWindow::RequestExitWithReturnCode(int return_code) {
+bool OrbitMainWindow::ExitWithWarning(int return_code) {
```

Comment:
Nit: From an API design perspective you should consider splitting up getting confirmation and exiting. (separation of concerns)

```cpp
bool OrbitMainWindow::ConfirmExit() const;
void OrbitMainwindow::Exit(int return_code);
```

Reply:
So `ConfirmExit()` just displays the warning? And then I call it something like this:
```
if (ConfirmExit()) Exit(0);
```
Did I get that right?

Reply:
Yes, exactly.

Reply:
I had to use `const_cast` in `ConfirmExit()` since `QMessageBox::warning()` is non const. I think this is a good use of `const_cast`, but I would like to know your opinion. The alternative would be to not make `ConfirmExit()` const.

Reply:
Ah I'm sorry for suggesting `const`. I would prefer to make it non-const.

You may be right that `const_cast` is okay here, but it sets a bad example, since there is really nothing that blocks us from making the method non-const.

Reply:
No worries, suggesting const made sense in the first place. And with this little detour I learned about `const_cast` and how to use it :). Removed it, will merge now.